diff -ru ../kirigami/src/controls/GlobalDrawer.qml ./src/controls/GlobalDrawer.qml
--- ../kirigami/src/controls/GlobalDrawer.qml	2019-10-10 17:13:41.384333413 -0700
+++ ./src/controls/GlobalDrawer.qml	2019-10-11 13:39:22.200854355 -0700
@@ -215,6 +215,20 @@
     readonly property Action currentSubMenu: stackView.currentItem ? stackView.currentItem.current: null
 
     /**
+     * scrollViewItem: ScrollView
+     *
+     * Points to the internal ScrollView so the caller can implement the Back button
+     */
+    readonly property Item scrollViewItem: scrollView
+
+    /**
+     * pop: function
+     *
+     * Point to the pop function of the stackView inside the scrollView in order to be able to implement the Back button in the caller
+     */
+    function pop() { scrollView.pop }
+
+    /**
      * Notifies that the banner has been clicked
      */
     signal bannerClicked()
@@ -238,6 +252,7 @@
         anchors.fill: parent
         implicitWidth: Math.min (Units.gridUnit * 20, root.parent.width * 0.8)
         horizontalScrollBarPolicy: Qt.ScrollBarAlwaysOff
+	function pop() { stackView.pop() }
 
         Flickable {
             id: mainFlickable
@@ -331,17 +346,13 @@
                 ColumnLayout {
                     id: topContent
                     spacing: 0
-                    Layout.alignment: Qt.AlignHCenter
-                    Layout.leftMargin: root.leftPadding
-                    Layout.rightMargin: root.rightPadding
                     Layout.bottomMargin: Units.smallSpacing
-                    Layout.topMargin: root.topPadding
                     Layout.fillWidth: true
                     Layout.fillHeight: true
                     Layout.preferredHeight: implicitHeight * opacity
                     //NOTE: why this? just Layout.fillWidth: true doesn't seem sufficient
                     //as items are added only after this column creation
-                    Layout.minimumWidth: parent.width - root.leftPadding - root.rightPadding
+		    Layout.minimumWidth: parent.width
                     visible: children.length > 0 && childrenRect.height > 0 && opacity > 0
                     opacity: !root.collapsed || showTopContentWhenCollapsed
                     Behavior on opacity {
@@ -435,7 +446,7 @@
                                 easing.type: Easing.InOutQuad
                             }
                         }
-
+/*
                         BasicListItem {
                             id: backItem
                             visible: level > 0
@@ -454,7 +465,7 @@
                             sequence: backItem.MnemonicData.sequence
                             onActivated: backItem.clicked()
                         }
-
+*/
                         Repeater {
                             id: actionsRepeater
                             model: root.actions
diff -ru ../kirigami/src/controls/private/GlobalDrawerActionItem.qml ./src/controls/private/GlobalDrawerActionItem.qml
--- ../kirigami/src/controls/private/GlobalDrawerActionItem.qml	2019-10-10 17:13:41.387666766 -0700
+++ ./src/controls/private/GlobalDrawerActionItem.qml	2019-10-11 14:16:03.013887752 -0700
@@ -94,7 +94,7 @@
         Layout.preferredHeight: !root.collapsed ? Units.iconSizes.smallMedium : Units.iconSizes.small/2
         selected: listItem.checked || listItem.pressed
         Layout.preferredWidth: Layout.preferredHeight
-        source: (LayoutMirroring.enabled ? "go-next-symbolic-rtl" : "go-next-symbolic")
+	source: (LayoutMirroring.enabled ? ":/go-next-symbolic-rtl" : ":/go-next-symbolic")
 		visible: (!isExpandible || root.collapsed) && !listItem.isSeparator && modelData.hasOwnProperty("children") && modelData.children!==undefined && modelData.children.length > 0
     }
     data: [
diff -ru ../kirigami/src/controls/private/globaltoolbar/PageRowGlobalToolBarUI.qml ./src/controls/private/globaltoolbar/PageRowGlobalToolBarUI.qml
--- ../kirigami/src/controls/private/globaltoolbar/PageRowGlobalToolBarUI.qml	2019-10-10 17:13:41.391000120 -0700
+++ ./src/controls/private/globaltoolbar/PageRowGlobalToolBarUI.qml	2019-10-11 13:39:22.204187697 -0700
@@ -40,6 +40,7 @@
     preferredHeight: globalToolBar.preferredHeight
     maximumHeight: globalToolBar.maximumHeight
     separatorVisible: globalToolBar.separatorVisible
+    background: Rectangle { color: subsurfaceTheme.primaryColor }
 
     RowLayout {
         anchors.fill: parent
@@ -110,6 +111,6 @@
             Layout.preferredWidth: height
         }
     }
-    background.opacity: breadcrumbLoader.opacity
+    //background.opacity: breadcrumbLoader.opacity
 }
 
diff -ru ../kirigami/src/controls/templates/private/BackButton.qml ./src/controls/templates/private/BackButton.qml
--- ../kirigami/src/controls/templates/private/BackButton.qml	2019-10-10 17:13:41.391000120 -0700
+++ ./src/controls/templates/private/BackButton.qml	2019-10-11 13:55:58.985325656 -0700
@@ -27,6 +27,7 @@
     id: button
 
     icon.name: (LayoutMirroring.enabled ? "go-previous-symbolic-rtl" : "go-previous-symbolic")
+    icon.source: (LayoutMirroring.enabled ? "go-previous-symbolic-rtl" : "go-previous-symbolic")
 
     enabled: applicationWindow().pageStack.layers.depth > 1 || (applicationWindow().pageStack.depth > 1 && (applicationWindow().pageStack.currentIndex > 0 || applicationWindow().pageStack.contentItem.contentX > 0))
 
diff -ru ../kirigami/src/controls/templates/private/ForwardButton.qml ./src/controls/templates/private/ForwardButton.qml
--- ../kirigami/src/controls/templates/private/ForwardButton.qml	2019-10-10 17:13:41.391000120 -0700
+++ ./src/controls/templates/private/ForwardButton.qml	2019-10-11 13:55:52.765272124 -0700
@@ -27,6 +27,7 @@
     id: button
 
     icon.name: (LayoutMirroring.enabled ? "go-next-symbolic-rtl" : "go-next-symbolic")
+    icon.source: (LayoutMirroring.enabled ? "go-next-symbolic-rtl" : "go-next-symbolic")
 
     enabled: applicationWindow().pageStack.currentIndex < applicationWindow().pageStack.depth-1
 
diff -ru ../kirigami/src/kirigamiplugin.cpp ./src/kirigamiplugin.cpp
--- ../kirigami/src/kirigamiplugin.cpp	2019-10-10 17:13:41.391000120 -0700
+++ ./src/kirigamiplugin.cpp	2019-10-11 13:39:22.204187697 -0700
@@ -41,7 +41,7 @@
 
 //Q_INIT_RESOURCE(kirigami);
 #ifdef KIRIGAMI_BUILD_TYPE_STATIC
-#include <qrc_kirigami.cpp>
+//#include <qrc_kirigami.cpp>
 #endif
 
 QUrl KirigamiPlugin::componentUrl(const QString &fileName) const
