diff -ur a/kirigami/kirigami.pri kirigami.pri
--- kirigami.pri	2018-08-24 13:27:24.000000000 -0700
+++ kirigami.pri	2018-09-27 09:02:36.000000000 -0700
@@ -5,6 +5,7 @@
                $$PWD/src/settings.h \
                $$PWD/src/formlayoutattached.h \
                $$PWD/src/mnemonicattached.h \
+               $$PWD/src/scenepositionattached.h \
                $$PWD/src/libkirigami/basictheme_p.h \
                $$PWD/src/libkirigami/platformtheme.h \
                $$PWD/src/libkirigami/kirigamipluginfactory.h \
@@ -16,6 +16,7 @@
                $$PWD/src/settings.cpp \
                $$PWD/src/formlayoutattached.cpp \
                $$PWD/src/mnemonicattached.cpp \
+               $$PWD/src/scenepositionattached.cpp \
                $$PWD/src/libkirigami/basictheme.cpp \
                $$PWD/src/libkirigami/platformtheme.cpp \
                $$PWD/src/libkirigami/kirigamipluginfactory.cpp \
diff -ur a/kirigami/src/controls/GlobalDrawer.qml src/controls/GlobalDrawer.qml
--- src/controls/GlobalDrawer.qml	2018-09-27 08:43:36.000000000 -0700
+++ src/controls/GlobalDrawer.qml	2018-09-27 09:01:41.000000000 -0700
@@ -328,17 +328,13 @@
                 ColumnLayout {
                     id: topContent
                     spacing: 0
-                    Layout.alignment: Qt.AlignHCenter
-                    Layout.leftMargin: root.leftPadding
-                    Layout.rightMargin: root.rightPadding
                     Layout.bottomMargin: Units.smallSpacing
-                    Layout.topMargin: root.topPadding
                     Layout.fillWidth: true
                     Layout.fillHeight: true
                     Layout.preferredHeight: implicitHeight * opacity
                     //NOTE: why this? just Layout.fillWidth: true doesn't seem sufficient
                     //as items are added only after this column creation
-                    Layout.minimumWidth: parent.width - root.leftPadding - root.rightPadding
+                    Layout.minimumWidth: parent.width
                     visible: children.length > 0 && childrenRect.height > 0 && opacity > 0
                     opacity: !root.collapsed || showTopContentWhenCollapsed
                     Behavior on opacity {
