diff --git a/libmtp.pc.in b/libmtp.pc.in
index 93c7bbf..ff817e1 100644
--- a/libmtp.pc.in
+++ b/libmtp.pc.in
@@ -10,6 +10,6 @@ Description: libmtp is a library for accessing Media Transfer Protocol devices
 Version: @VERSION@
 Requires.private: @LIBUSB_REQUIRES@
 Conflicts:
-Libs: -L${libdir} -lmtp
+Libs: -L${libdir} -lmtp @OSFLAGS@
 Libs.private: @LIBS@
-Cflags: -I${includedir} @OSFLAGS@
+Cflags: -I${includedir}
