From 5e745f7a02fd967dbfabeba0871025b1513ce1d4 Mon Sep 17 00:00:00 2001
From: Dirk Hohndel <dirk@hohndel.org>
Date: Wed, 25 Nov 2020 13:51:35 -0800
Subject: [PATCH 05/11] breadcrumbs: get better font size

Signed-off-by: Dirk Hohndel <dirk@hohndel.org>
---
 src/controls/private/globaltoolbar/BreadcrumbControl.qml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/controls/private/globaltoolbar/BreadcrumbControl.qml b/src/controls/private/globaltoolbar/BreadcrumbControl.qml
index ad80d222..c45db280 100644
--- a/src/controls/private/globaltoolbar/BreadcrumbControl.qml
+++ b/src/controls/private/globaltoolbar/BreadcrumbControl.qml
@@ -69,6 +69,7 @@ Flickable {
                     }
                     Kirigami.Heading {
                         Layout.leftMargin: Kirigami.Units.largeSpacing
+                        font: Kirigami.Theme.defaultFont
                         color: Kirigami.Theme.textColor
                         verticalAlignment: Text.AlignVCenter
                         wrapMode: Text.NoWrap
-- 
2.25.1

